cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/kubernetes/AbstractNamespacedResourceOperatorTest.java

```diff
@@ -117,17 +123,40 @@ protected T modifiedResource()  {
     /** Create the subclass of ResourceOperation to be tested */
     protected abstract AbstractNamespacedResourceOperator<C, T, L, R> createResourceOperations(Vertx vertx, C mockClient);
 
-    /** Create the subclass of ResourceOperation to be tested with mocked readiness checks*/
-    protected AbstractNamespacedResourceOperator<C, T, L, R> createResourceOperationsWithMockedReadiness(Vertx vertx, C mockClient)    {
+    /**
+     *  Create the subclass of ResourceOperation to be tested.
+     *  The boolean parameter useServerSideApply is not used by default.
+     *  In case that resource supports SSA, the method should be overridden in resource's test class
+     */
+    protected AbstractNamespacedResourceOperator<C, T, L, R> createResourceOperations(Vertx vertx, C mockClient, boolean useServerSideApply) {
```

Comment:
This does not seem to make sense. Why does it just loose the `useServerSideApply` param? I guess that suggests that you do not really need two methods and just one of them?

Reply:
It was the easiest solution how to not change all the classes for resources that don't bother with SSA.
In case that the particular resource has possibility to enable SSA, it will have this method overwritten with correct implementation anyway.

Reply:
In case that we want just one base for the tests and for the parametrized tests (which makes sense to me as otherwise there would be a lot of duplication), I would keep it.
But possibly, I would add something like this:

```
if (useServerSideApply && !supportsServerSideApply()) {
        throw new UnsupportedOperationException("This resource does not support server-side apply");
}
```

into it and also I would add boolean method `supportsServerSideApply`, defaulting to `false`, which can be then overwritten in resource classes that support SSA.

Reply:
Also with `supportsServerSideApply` I could then just add some UTs for the update with conflicts (so check if the `force` was applied) into the base class, without need to create another base class just for SSA.